File: Java/03- Interface/03- Funcionário/Funcionario.java

package com.mycompany.mavenproject5;

public abstract class Funcionario {
    
    protected String nome;
    protected String dataNacimento;
    protected Setor setor;        
    protected Sexo sexo;
    protected double salarioBase;

    public Funcionario(String dataNacimento, String nome, double salarioBase, Setor setor, Sexo sexo) {
        this.dataNacimento = dataNacimento;
        this.nome = nome;
        this.salarioBase = salarioBase;
        this.setor = setor;
        this.sexo = sexo;
    }

    public String getDataNacimento() {
        return dataNacimento;
    }

    public void setDataNacimento(String dataNacimento) {
        this.dataNacimento = dataNacimento;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public double getSalarioBase() {
        return salarioBase;
    }

    public void setSalarioBase(double salarioBase) {
        this.salarioBase = salarioBase;
    }

    public Setor getSetor() {
        return setor;
    }

    public void setSetor(Setor setor) {
        this.setor = setor;
    }

    public Sexo getSexo() {
        return sexo;
    }

    public void setSexo(Sexo sexo) {
        this.sexo = sexo;
    }

    //A Implementação não é obrigatória.
    public abstract double getSalarioFinal();
    
    
    @Override
    public String toString() {
        return 
                "\nNome: " + nome + 
                "\nData de Nacimento: " + dataNacimento + 
                "\nSetor: " + setor.getNome() + 
                "\nSexo: " + sexo.name() + 
                "\nSalário Base: " + salarioBase +
                "\nSaçário Final: " + getSalarioFinal();
    }
    
}

File: Python/04- POO/02- Com pastas/02- Atividade/models/funcionario.py
from models.enum.sexo import Sexo
from models.enum.setor import Setor

class Funcionario: 
    def __init__(self, id: int, nome:  str, idade: int, salario : float, sexo: Sexo, setor: Setor) -> None:
        self.id = id
        self.nome = nome 
        self.idade = idade
        self.salario = salario
        self.sexo = sexo
        self.setor = setor

    def __str__(self) -> str:
        return (
                f"Id: {self.id}"
                f"\nNome: {self.nome}"
                f"\nIdade: {self.idade} Anos"
                f"\nSalário: {self.salario} R$"
                f"\nSexo: {self.sexo.value}"
                f"\nSetor: {self.setor.value}"
                )    